EpikCord/__init__.py

```diff
@@ -4186,4 +4188,90 @@ async def default_failure(self, interaction):
 class CommandUtils:
     
     def check(self, callback):
-        return Check(callback)
\ No newline at end of file
+        return Check(callback)
+
+class BaseConnectable:
+    def __init__(self,client:Client):
+        self.client = client
+        self.ws = None
+        self.IDENTIFY = 0
+        self.SELECT_PROTOCOL = 1
+        self.READY = 2
+        self.HEARTBEAT = 3
+        self.HELLO = 8
+        self.connected = False
+        self.server_set = False
+        self.state_set = False
+        self.sequence = None
+        self.endpoint:str = None
+        #have to add events manually
+        self.client.events["voice_server_update"] = self.voice_server_update
+        self.client.events["voice_state_update"] = self.voice_state_update
+    async def voice_state_update(self,data):
+        if self.connected and data["channel_id"]== self.channel_id:
+            self.session_id = data["session_id"]
+            self.state_set = True
+
+
+    async def voice_server_update(self, data):
+        if self.connected and data["guild_id"] == self.guild_id:
+            self.token = data["token"]
+            self.guild_id = data["guild_id"] #ain't gonna make a difference
+            self.endpoint = data["endpoint"]
+            self.server_set = True
+    async def connect(self, guild_id:str, channel_id:str, self_mute:bool = False, self_deaf:bool = False):
+        
+        await self.client.send_json({
+            "op":4,
+            "d":{
+                "guild_id":guild_id,
+                "channel_id":channel_id,
+                "self_mute":self_mute,
+                "self_deaf": self_deaf
+            }
+
+        })
```

Comment:
Maybe extract the payload logic for readability

Reply:
wdym

Reply:
I believe this should be done in another pr, but function that return payload wouldn't add any latency while getting all the json payload outside the class logic.
So like
```py
await self.send_json(prepare_payload(PAYLOAD_X, payload_item)
```